backend/apps/matching/admin.py

```diff
@@ -23,6 +23,42 @@ def participant_required(function=None, redirect_field_name=REDIRECT_FIELD_NAME,
     return actual_decorator
 
 
+def can_approve_type_in_url(function=None):
+    """
+    Check that the user at hand can approve objects of given p_type
+    """
+
+    def actual_decorator(function):
+        def new_func(request, p_type, *args, **kwargs):
+            if request.user.has_perm(p_type):
+                return function(request, p_type, *args, **kwargs)
+            raise Http404
+
+        return new_func
+
+    if function:
+        return actual_decorator(function)
+    return actual_decorator
+
+
+def can_delete_type_in_url(function=None):
```

Comment:
I'm not sure anymore whether decorators with this much logic make sense, it depends on the type of user and a permission dynamically.
And as I, a member of this team and very familiar with this specific piece of code, took almost a minute to figure out what `in_url` means, I am unsure about it even more... I first thought it was another permission stating where you are allowed to delete users.